pkg/model/time_series_merger.go

```diff
@@ -20,6 +22,40 @@ func MergeSeries(aggregation *typesv1.TimeSeriesAggregationType, series ...[]*ty
 	return m.TimeSeries()
 }
 
+// TopSeries returns the top k series by sum of values.
+// If k is zero, all series are returned.
+// Note that even if len(c) <= k or k == 0, the returned
+// series are sorted by value in descending order and then
+// lexicographically (in ascending order).
+func TopSeries(s []*typesv1.Series, k int) []*typesv1.Series {
+	type series struct {
+		*typesv1.Series
+		sum float64
+	}
+	aggregated := make([]series, len(s))
+	for i, x := range s {
+		var sum float64
+		for _, p := range x.Points {
+			sum += p.Value
+		}
+		aggregated[i] = series{Series: x, sum: sum}
+	}
+	slices.SortFunc(aggregated, func(a, b series) int {
+		c := cmp.Compare(a.sum, b.sum)
+		if c == 0 {
+			return CompareLabelPairs(a.Labels, b.Labels)
+		}
+		return c * -1 // Invert to sort in descending order.
```

Comment:
This is strictly an observation but to invert a number you can use the negative sign as well.

```suggestion
		return -c // Invert to sort in descending order.
```